.github/workflows/ci.yml

```diff
@@ -266,3 +266,94 @@ jobs:
       uses: codecov/codecov-action@v2
       with:
         files: ${{github.workspace}}/source/coverage.info
+
+#----------------------------------------------------------------------------
+# Sanitizer: Build and test on linux with last VTK with sanitizer options
+#----------------------------------------------------------------------------
+  sanitizer:
+    if: github.event.pull_request.draft == false
+
+    strategy:
+      fail-fast: false
+      matrix:
+        # "memory" is not supported by the GCC version in our Docker image yet
+        # "leak" is run by defaut with "address" so we do not need to run it in CI
+        santype: [address, thread, undefined]
+
+    runs-on: ubuntu-latest
+    container: ghcr.io/f3d-app/f3d-ci
+
+    env:
+      DISPLAY: :0
+      LSAN_OPTIONS: suppresions=${{github.workspace}}/source/.lsan.supp
+
+    steps:
+
+    - name: Checkout
+      uses: actions/checkout@v2
+      with:
+        path: 'source'
+        fetch-depth: 0
+        lfs: 'false'
+
+    - name: Set LFS env var
+      working-directory: ${{github.workspace}}/source
+      shell: bash
+      run: echo "LFS_SHA=$(git log -n 1 --pretty=format:%H -- testing/data/ testing/baselines/)" >> $GITHUB_ENV
+
+    - name: Copy LFS Data
+      uses: ./source/.github/actions/lfs-copy
+      with:
+        lfs_sha: ${{env.LFS_SHA}}
+
+    - name: Dependencies Dir
+      working-directory: ${{github.workspace}}
+      run: mkdir dependencies
+
+    - name: Install OCCT dependency
+      uses: ./source/.github/actions/occt-install-dep
+
+    - name: Install Assimp dependency
+      uses: ./source/.github/actions/assimp-install-dep
+
+    - name: Install Imath dependency
+      uses: ./source/.github/actions/imath-install-dep
+
+    - name: Install Alembic dependency
+      uses: ./source/.github/actions/alembic-install-dep
+
+    - name: Install VTK dependency
+      uses: ./source/.github/actions/vtk-install-dep
+      with:
+        vtk_sha_file: ./source/.github/actions/vtk-install-dep/vtk_commit_sha
+
+    # pwd seems necessary for OpenCASCADE and Assimp for some reasons
+    # coverage build is done in source as it seems to be required for codecov
```

Comment:
this is not coverage, and is in source build needed for lsan ?

Reply:
fixed